.github/workflows/rubocop.yml

```diff
@@ -43,23 +34,46 @@ jobs:
         run: ruby -I lib -r bundler/setup -r rubocop -e 'exit 0'
       - name: Set up Coverage
         # Only collect coverage data on ubuntu runners with CRuby
-        if: matrix.os == 'ubuntu' && !startsWith(matrix.ruby, 'jruby-')
+        if: ${{ !startsWith(matrix.ruby, 'jruby-') }}
         run: echo "COVERAGE=true" >> $GITHUB_ENV
       - name: spec
         env:
           CI_RUBY_VERSION: ${{ matrix.ruby }}
         run: bundle exec rake spec
       - name: Upload Coverage Artifact
-        if: matrix.os == 'ubuntu' && !startsWith(matrix.ruby, 'jruby-')
+        if: ${{ !startsWith(matrix.ruby, 'jruby-') }}
         uses: actions/upload-artifact@v4
         with:
-          name: coverage-${{ matrix.os }}-${{ matrix.ruby }}
+          name: coverage-ubuntu-${{ matrix.ruby }}
           path: coverage/.resultset.json
           if-no-files-found: error
 
+  spec-windows:
```

Comment:
I'd add some comments here about the reasoning for running the Windows tests after the Ubuntu tests, so someone won't accidentally change this in the future.